### PR TITLE
Added citations to the relevant papers for Adam, AdaGrad, and RMSprop

### PR DESCRIPTION
**Pull request by zackchase**
Added citations to the relevant papers for Adam, AdaGrad, and RMSprop keeping with the documentation style used in the RNN docstrings. I think this would be helpful, especially for lesser known optimizers like Adam.

### CODE REVIEW

File: tensorflow/python/training/adam.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
 
 class AdamOptimizer(optimizer.Optimizer):
-  """Optimizer that implements the Adam algorithm.
+  """Optimizer that implements the Adam algorithm (http://arxiv.org/abs/1412.6980).
```
Comment (rafaljozefowicz) [inline tensorflow/python/training/adam.py]: Line 41 references the paper.

**Comment by zackchase** [comment]
I signed it!

File: tensorflow/python/training/adam.py (hunk 1 above)
Comment (zackchase) [inline tensorflow/python/training/adam.py]: Sorry didn't see it. Should the ref be in init or in the class docstring?

Elsewhere, the arxiv reference is in the class docstring:

class GRUCell(RNNCell):"""Gated Recurrent Unit cell (cf.
http://arxiv.org/abs/1406.1078)."""

Comment (rafaljozefowicz) [inline tensorflow/python/training/adam.py]: I think what you have looks good. Can you remove that other reference and maybe point to specific version of the paper (v7)? The formulas were updated from previous versions.

Comment (zackchase) [inline tensorflow/python/training/adam.py]: Done! Pointing to v7 and removed the duplicate reference.

**Comment by vrv** [comment]
Cool, squash the commits into one and we'll merge it in.  Thanks!

**Comment by zackchase** [comment]
No problem. Done! Should appear as one commit now.

**Pull request merged**